### DISCUSSION TITLE
How to make lines look hand drawn like manga/Comic?

### DISCUSSION THREAD

**Discussion by krigeta** [bnpr/Malt#305] [Q&A] [answered]
Hello, Is there any way to make the line look more hand-drawn and less straight? like the outline of characters in comics or manga?

**Comment by eballeste** [comment]
I was wondering something very similar, how can we accomplish custom outlines, is there something like illustrator's stroke profiles?
 
my wish would be to accomplish stylistic outlines that look like this, like if it was done with markers that had gaps in it:
![Screen Shot 2022-05-25 at 8 41 20 PM](https://user-images.githubusercontent.com/6924108/170392256-cd588b6b-2c9f-42a2-98ca-97d5e3aaa204.png)

**Reply by krigeta** [reply]
> I was wondering something very similar, how can we accomplish custom outlines, is there something like illustrator's stroke profiles?
> 
> my wish would be to accomplish stylistic outlines that look like this, like if it was done with markers that had gaps in it: ![Screen Shot 2022-05-25 at 8 41 20 PM](https://user-images.githubusercontent.com/6924108/170392256-cd588b6b-2c9f-42a2-98ca-97d5e3aaa204.png)

you can check this for this, but for me it is not working(by the developer himself)
![134017278-1ebb78da-b720-4f6f-9d44-224ce01060c2](https://user-images.githubusercontent.com/75309361/170418077-14d8757a-6d08-4e01-ada8-810c715adee7.png)

**Comment by pragma37** [comment, marked as answer]
> I was wondering something very similar, how can we accomplish custom outlines, is there something like illustrator's stroke profiles?

@eballeste Lines in Malt are pixel-based so you can't control them as if they were vectors like in Freestyle and LineArt.

You can, however, drive the line width and color with any shader value (textures, vertex colors, lighting, procedural noise...). You can even post-process the line properties before rendering them in the RenderLayer nodes.

> you can check this for this, but for me it is not working(by the developer himself)

@krigeta That's using pre-1.0 nodes. Here's a basic up to date example for randomizing line width:

![imagen](https://user-images.githubusercontent.com/36610999/170717354-1a4957cd-7d0e-4ab3-a33d-de9ba713a876.png)

[hand_drawn_lines.zip](https://github.com/bnpr/Malt/files/8787296/hand_drawn_lines.zip)
